### ISSUE TITLE
Failed to terminate kernel

### ISSUE THREAD

**Issue by christianu7** [nteract/hydrogen#1756]
[Enter steps to reproduce:]

1. Open two atom sessions using "atom ." in the command line
1. initialize python 3 in one of them
2. Change working directory using
   import os
   os.chdir(script_dir)
3. Shutdown python kernel
4. Uncaught Error: ENOENT: no such file or directory, unlink 

**Atom**: 1.40.1 x64
**Electron**: 3.1.10
**OS**: Mac OS X 10.14.6
**Thrown From**: [Hydrogen](https://github.com/nteract/hydrogen) package 2.12.0

### Stack Trace

Uncaught Error: ENOENT: no such file or directory, unlink '/Users/Chris/Library/Jupyter/runtime/kernel-a22fc0a3-e9fd-42f3-8dee-88f556363d74.json'

```
At fs.js:143

Error: ENOENT: no such file or directory, unlink '/Users/Chris/Library/Jupyter/runtime/kernel-a22fc0a3-e9fd-42f3-8dee-88f556363d74.json'
    at Object.fs.unlinkSync (fs.js:1061:3)
    at ZMQKernel.destroy (/packages/Hydrogen/lib/zmq-kernel.js:393:8)
    at Kernel.destroy (/packages/Hydrogen/lib/kernel.js:435:20)
    at /packages/Hydrogen/lib/main.js:300:14)
    at Object.handleKernelCommand (/packages/Hydrogen/lib/main.js:310:4)
    at SignalListView.signalListView.onConfirmed (/packages/Hydrogen/lib/services/consumed/status-bar/status-bar.js:59:14)
    at Object.didConfirmSelection (/packages/Hydrogen/lib/services/consumed/status-bar/signal-list-view.js:40:36)
    at SelectListView.confirmSelection (/packages/Hydrogen/node_modules/atom-select-list/src/select-list-view.js:400:20)
    at SelectListView.didClickItem (/packages/Hydrogen/node_modules/atom-select-list/src/select-list-view.js:286:10)
    at ListItemView.onclick (/packages/Hydrogen/node_modules/atom-select-list/src/select-list-view.js:221:31)
    at ListItemView.didClick (/packages/Hydrogen/node_modules/atom-select-list/src/select-list-view.js:448:10)
```

### Commands

```
     -0:50.5.0 core:confirm (input.hidden-input)
     -0:26.9.0 core:move-to-top (input.hidden-input)
     -0:26.3.0 editor:select-to-end-of-line (input.hidden-input)
     -0:24.6.0 hydrogen:run (input.hidden-input)
     -0:24.6.0 autocomplete-plus:cancel (atom-text-editor.editor.is-focused.has-selection)
     -0:24.2.0 core:move-down (input.hidden-input)
     -0:23.9.0 core:confirm (input.hidden-input)
     -0:16.8.0 core:move-down (input.hidden-input)
  3x -0:16.5.0 core:select-down (input.hidden-input)
     -0:15.9.0 hydrogen:run (input.hidden-input)
     -0:15.9.0 autocomplete-plus:cancel (atom-text-editor.editor.is-focused.has-selection)
     -0:13.4.0 core:move-down (input.hidden-input)
     -0:13.1.0 core:move-up (input.hidden-input)
     -0:12.7.0 core:select-to-top (input.hidden-input)
     -0:12.4.0 hydrogen:run (input.hidden-input)
     -0:12.4.0 autocomplete-plus:cancel (atom-text-editor.editor.is-focused.has-selection)
```

### Non-Core Packages

```
Hydrogen 2.12.0 
language-latex 1.2.0 
language-markdown 0.37.0 
latex 0.50.2 
latex-completions 0.3.6 
pdf-view 0.71.0 
pen-paper-coffee-syntax 0.17.0 
```

**Comment by piovac** [comment]
I have the same issue. It is intermittent though.

Also, I used to have other two issues with kernels that disappeared since this new problem appeared, are maybe they are related related:
- If I quit Atom without shutting down the kernel first, many times the kernels stayed alive and I needed to kill their processes manually with a shell kill command
- the `runtime` library was literally full of hundreds of UUID files, meaning that the kernel shutdown was not clean.

I guess the shutdown process of the kernel need to be reviewed.
